packages/medusa-plugin-auth/src/auth-strategies/auth0/__tests__/store/verify-callback.spec.ts

```diff
@@ -0,0 +1,161 @@
+								return {
+									id: 'test',
+								};
+							}
+
+							if (email === existsEmailWithMeta) {
+								return {
+									id: 'test2',
+									metadata: {
+										[CUSTOMER_METADATA_KEY]: true,
+									},
+								};
+							}
+
+							if (email === existsEmailWithMetaAndProviderKey) {
+								return {
+									id: 'test3',
+									metadata: {
+										[CUSTOMER_METADATA_KEY]: true,
+										[AUTH_PROVIDER_KEY]: AUTH0_STORE_STRATEGY_NAME
+									},
+								};
+							}
+
+							if (email === existsEmailWithMetaButWrongProviderKey) {
+								return {
+									id: 'test4',
+									metadata: {
+										[CUSTOMER_METADATA_KEY]: true,
+										[AUTH_PROVIDER_KEY]: 'fake_provider_key'
+									},
+								};
+							}
+
+							return;
+						}),
+					},
+				};
+
+				return container_[name];
+			},
+		} as MedusaContainer;
+
+		auth0StoreStrategy = new Auth0StoreStrategy(
+			container,
+			{} as ConfigModule,
+			{ auth0Domain: 'fakeDomain', clientID: 'fake', clientSecret: 'fake', store: { callbackUrl: '/fakeCallbackUrl'} } as Auth0Options
+		);
+	});
+
+	afterEach(() => {
+		jest.clearAllMocks();
+	});
+
+	it('should succeed', async () => {
+		profile = {
+			emails: [{ value: existsEmailWithMetaAndProviderKey }],
+		};
+
+		const data = await auth0StoreStrategy.validate(req, accessToken, refreshToken, extraParams, profile);
+		expect(data).toEqual(
+			expect.objectContaining({
+				id: 'test3',
+			})
+		);
+	});
+
+	it('should fail when the customer exists without the metadata', async () => {
+		profile = {
+			emails: [{ value: existsEmail }],
+		};
+
+		const err = await auth0StoreStrategy.validate(req, accessToken, refreshToken, extraParams, profile).catch((err) => err);
+		expect(err).toEqual(new Error(`Customer with email ${existsEmail} already exists`));
+	});
+
+	it('should update customer metadata when the customer exsits with ONLY customer metadata key', async () => {
+		profile = {
+			emails: [{ value: existsEmailWithMeta }],
+		};
+
+    const data = await auth0StoreStrategy.validate(req, accessToken, refreshToken, extraParams, profile);
+		expect(data).toEqual(
+			expect.objectContaining({
+				id: 'test2',
+			})
+		);
+	});
+
+	it('should fail when the metadata exists but auth provider key is wrong', async () => {
+		profile = {
+			emails: [{ value: existsEmailWithMetaButWrongProviderKey }],
+		};
+
+		const err = await auth0StoreStrategy.validate(req, accessToken, refreshToken, extraParams, profile).catch((err) => err);
+		expect(err).toEqual(new Error(`Customer with email ${existsEmailWithMetaButWrongProviderKey} already exists`));
+	});
+
+	it('should succeed and create a new customer if it has not been found', async () => {
+		profile = {
+			emails: [{ value: 'fake' }],
+			name: {
+				givenName: 'test',
+				familyName: 'test',
+			},
+		};
+
+		const data = await auth0StoreStrategy.validate(req, accessToken, refreshToken, extraParams, profile);
+		expect(data).toEqual(
```

Comment:
todo: Add a check that the create method have been called here. also with the right input data

Reply:
+1 Jest not allowing me to mock the `update` function